SpatialGDK/Source/SpatialGDK/Private/Utils/ActorGroupManager.cpp

```diff
@@ -0,0 +1,202 @@
+#include "Utils/ActorGroupManager.h"
+#include "SpatialGDKSettings.h"
+
+UActorGroupManager::UActorGroupManager()
+{
+	InitFromSettings();
+}
+
+UActorGroupManager* UActorGroupManager::GetInstance()
+{
+	UActorGroupManager* Manager = NewObject<UActorGroupManager>();
+	Manager->InitFromSettings();
+	return Manager;
+}
+
+void UActorGroupManager::InitFromSettings()
+{
+	const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>();
+
+	ClassPathToActorGroup.Empty();
+	ActorGroupToWorkerType.Empty();
+
+	auto ActorGroups = DefaultActorGroups();
+	auto WorkerAssociation = DefaultWorkerAssociation();
+	if (Settings->bEnableOffloading)
+	{
+		ActorGroups = Settings->ActorGroups;
+		WorkerAssociation = Settings->WorkerAssociation.ActorGroupToWorker;
+	}
+
+	for (TPair<FName, FActorClassSet> ActorGroup : ActorGroups)
+	{
+		for (TSoftClassPtr<AActor> ClassPtr : ActorGroup.Value.ActorClasses)
+		{
+			ClassPathToActorGroup.Add(ClassPtr, ActorGroup.Key);
+		}
+	}
+
+	ActorGroupToWorkerType.Append(WorkerAssociation);
+}
+
+FName UActorGroupManager::GetActorGroupForClass(UClass* Class)
```

Comment:
I want to see an easy way to check if the type of the running worker is associated with the actor group of a given actor. This is by far the most common check that needs to be done when porting game code. I envision this as either a new member of `AActor` or a static member / member which is part of a new "running worker" abstraction. So far this information was available in the game instance, are you planning to keep this design?

Reply:
Can the param be marked as const?